saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/Saml2AuthenticationRequest.java

```diff
@@ -96,12 +104,38 @@ public String getAssertionConsumerServiceUrl() {
 		return this.credentials;
 	}
 
+	/**
+	 * Returns true if this AuthNRequest will be sent using REDIRECT binding and
+	 * encoded with the DEFLATE encoding. False if the AuthNRequest will be sent using POST binding.
+	 * @return true if REDIRECT/DEFLATE binding/encoding should be used. false for POST binding.
+	 */
+	public boolean isDeflate() {
+		return this.deflate;
+	}
+
+	/**
+	 * Returns an ordered map of query parameters to be used in the AuthNRequest
+	 * @return an ordered map of query parameters to be used in the AuthNRequest
+	 */
+	public MultiValueMap<String, String> getQueryParameters() {
+		return this.queryParameters;
+	}
+
 	/**
 	 * A builder for {@link Saml2AuthenticationRequest}.
 	 * returns a builder object
 	 */
 	public static Builder builder() {
-		return new Builder();
+		return new Builder(null);
+	}
+
+	/**
+	 * A builder for {@link Saml2AuthenticationRequest}.
+	 * returns a builder object with prepopulated data
+	 * @param request - the data to be prepopulated in the builder object
+	 */
+	public static Builder builder(Saml2AuthenticationRequest request) {
```

Comment:
A common Spring Security convention is to do `withFoo(Foo foo)` instead of `builder`. We use builder with nothing at all is provided in the static method.

In this case, it would be closer to the rest of Spring Security to do `withAuthenticationRequest`.

Note, though, that `OAuth2AuthorizationRequest`, which feels like the OAuth 2.0 analog for this class, does `from(OAuth2AuthorizationRequest)` for its copy-constructing static factory method.

Reply:
I agree, please consider f5c9769b0cf5c941c6efd2509ad3b5d4bb46f087 as the solution